packages/design-system/src/components/datalist/datalist.js

```diff
@@ -63,8 +63,7 @@ const Container = styled.div`
  * @param {Function} props.onObserve When this is present, observer will detect new options coming into view and trigger the function for these entries.
  * @param {Array} props.primaryOptions Array of options to display by default when not searching
  * @param {string} props.primaryLabel Label to display above the primary options.
- * @param {Array} props.priorityOptions Options to display in front of all the other options in a separate group (will not remove these from the `options`).
- * @param {string} props.priorityLabel Label to display in front of the priority options.
+ * @param {Array} props.priorityOptionGroups Groups of pptions to display in front of all the other options (will not remove these from the `options`).
```

Comment:
```suggestion
 * @param {Array} props.priorityOptionGroups Groups of options to display in front of all the other options (will not remove these from the `options`).
```

i like that you're combining the options and labels into groups, good refactor!